lib/dns.js

```diff
@@ -192,10 +195,10 @@ function onlookupservice(err, host, service) {
 // lookupService(address, port, callback)
 exports.lookupService = function(host, port, callback) {
   if (arguments.length !== 3) {
-    throw new Error('Invalid arguments');
+    throw new TypeError(`Expected 3 arguments, got ${arguments.length}`);
```

Comment:
This isn't really a TypeError per se, since it is more to do with argument count rather than argument type.